sound/soc/sof/topology.c

```diff
@@ -1106,6 +1106,7 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 		dai->name = rtd->dai_link->name;
 		dev_dbg(scomp->dev, "tplg: connected widget %s -> DAI link %s\n",
 			w->name, rtd->dai_link->name);
+		break;
```

Comment:
But don't we support joint/split pipelines? Or they needs to be connected at a module copier?
```
Playback PCM1 ---\
                  ---> Playback DAI
Playback PCM1 ---/


                /--- Capture PCM1
Capture DAI ----
                \--- Capture PCM2
```

Reply:
@ujfalusi I wondered about this, but not sure how the existing code would handle this, e,g. the "dai->name = rtd->dai_link->name" assignment. This would be assigned to last matching DAI link. Seems a bit random.

Reply:
right, true, so from last to first?

Reply:
> But don't we support joint/split pipelines? Or they needs to be connected at a module copier?
> 
> ```
> Playback PCM1 ---\
>                   ---> Playback DAI
> Playback PCM1 ---/
> 
> 
>                 /--- Capture PCM1
> Capture DAI ----
>                 \--- Capture PCM2
> ```

@ujfalusi even with the connected pipelines , there's still only 1 DAI link isnt it? What would create an issue here? We already support the mixed playback PCM into a single DAI case in our topologies today

Reply:
this is not in a hot path, so, what is we do this:
scan the dai links with partial match
If only a single match is found, go with it, if multiple ones found (when we face with the second match) then we re-run with an exact match?

Reply:
Now implemented this, please see V2.